packages/omg-js-util/src/transaction.js

```diff
@@ -294,6 +333,21 @@ function validateOutputs (arg) {
   }
 }
 
+function validateMetadata (arg) {
+  if (arg) {
+    let invalid = false
+    try {
+      const hex = Buffer.from(arg.replace('0x', ''), 'hex')
+      invalid = hex.length !== 32
+    } catch (err) {
+      invalid = true
+    }
+    if (invalid) {
+      throw new InvalidArgumentError('Metadata must be a 32-byte hex string')
```

Comment:
this should stop people from shooting themselves in the foot with faulty metadata